monai/networks/nets/dints.py

```diff
@@ -0,0 +1,556 @@
+                )
+
+        # define NAS search space
+        if code is None:
+            code_a = np.ones((num_blocks, len(code2out)))
+            code_c = np.ones((num_blocks, len(code2out), cell_ops))
+        else:
+            code_a = code[1]
+            code_c = F.one_hot(torch.from_numpy(code[2]), cell_ops).numpy()
+        self.cell_tree = nn.ModuleDict()
+        self.memory = np.zeros((num_blocks, len(code2out), cell_ops))
+        for blk_idx in range(num_blocks):
+            for res_idx in range(len(code2out)):
+                if code_a[blk_idx, res_idx] == 1:
+                    self.cell_tree[str((blk_idx, res_idx))] = cell(
+                        filter_nums[code2in[res_idx] + int(use_stem)],
+                        filter_nums[code2out[res_idx] + int(use_stem)],
+                        code2ops[res_idx],
+                        code_c[blk_idx, res_idx],
+                    )
+                    self.memory[blk_idx, res_idx] = np.array(
+                        [
+                            _.memory + self.cell_tree[str((blk_idx, res_idx))].preprocess.memory if _ is not None else 0
+                            for _ in self.cell_tree[str((blk_idx, res_idx))].op._ops[:cell_ops]
+                        ]
+                    )
+
+        # define cell and macro arhitecture probabilities
+        self.log_alpha_c = torch.nn.Parameter(
+            torch.zeros(num_blocks, len(code2out), cell_ops).normal_(1, 0.01).cuda().requires_grad_()
+        )
+        self.log_alpha_a = torch.nn.Parameter(
+            torch.zeros(num_blocks, len(code2out)).normal_(0, 0.01).cuda().requires_grad_()
+        )
+        self._arch_param_names = ["log_alpha_a", "log_alpha_c"]
+
+    def weight_parameters(self):
+        return [param for name, param in self.named_parameters() if name not in self._arch_param_names]
+
+    def get_prob_a(self, child: bool = False):
+        """
+        Get final path probabilities and child model weights
+
+        Args:
+            child: return child probability as well (used in decode)
+        """
+        log_alpha = self.log_alpha_a
+        _code_prob_a = torch.sigmoid(log_alpha)
+        norm = 1 - (1 - _code_prob_a).prod(-1)  # normalizing factor
+        code_prob_a = _code_prob_a / norm.unsqueeze(1)
+        if child:
+            probs_a = []
+            path_activation = torch.from_numpy(self.child_list).cuda()
+            for blk_idx in range(self.num_blocks):
+                probs_a.append(
+                    (
+                        path_activation * _code_prob_a[blk_idx] + (1 - path_activation) * (1 - _code_prob_a[blk_idx])
+                    ).prod(-1)
+                    / norm[blk_idx]
+                )
+            probs_a = torch.stack(probs_a)
+            return probs_a, code_prob_a
+        else:
+            return None, code_prob_a
+
+    def _get_memory_usage(self, in_size, cell_memory: bool = False, code: bool = None, full: bool = False):
```

Comment:
I feel maybe it's better to use `buffer` or other word instead of `memory`?

Thanks.